src/Middleware/HttpOverrides/src/ForwardedHeadersMiddleware.cs

```diff
@@ -167,14 +167,31 @@ public void ApplyForwarders(HttpContext context)
             entryCount = Math.Max(forwardedHost.Length, entryCount);
         }
 
+        if (_options.ForwardedHeaders.HasFlag(ForwardedHeaders.XForwardedPort))
+        {
+            checkPort = true;
+            forwardedPort = requestHeaders.GetCommaSeparatedValues(_options.ForwardedPortHeaderName);
+            if (_options.RequireHeaderSymmetry
+                && ((checkFor && forwardedFor!.Length != forwardedPort.Length)
```

Comment:
I'll be honest and say that I don't know what header symmetry is (it appears to require some amount of similarity between parallel headers), but it's not obvious to me that this check is correct.  For one thing, the old pattern appears to be that the first block made 1 check, the second block made 2 checks, and the third block made 3 checks.  Now, for some reason, the third block makes 4 checks.

Reply:
Oh, the last check is comparing `forwardedPort!.Length` to itself - that's probably not necessary.  That, at least, gets us back to a linearly increasing number of checks.